Removes deprecated methods

sgiehl opened the pull request:
As tests are currently failing due to not removed deprecated methods, I had a quick look and removed the methods we had tests for.

I only didn't remove/replace the method `Plugin::getListHooksRegistered()` as the removal was announced for Piwik 4 in changelog. Have adjusted the test in that case. Or shall we remove that for Piwik 3?

Also opened https://github.com/piwik/plugin-TasksTimetable/pull/10 and https://github.com/piwik/plugin-SecurityInfo/pull/17 to fix the plugins as they were still using `Menu::add()`

sgiehl committed:
Removes deprecated method MenuAbstract::add

sgiehl committed:
Removes deprecated method Archive::getDataTableFromArchive

sgiehl committed:
method Plugin::getListHooksRegistered() was announced to be removed in Piwik 4 in CHANGELOG

sgiehl committed:
Removes deprected renameUserSettingsModuleAndAction() methods

sgiehl committed:
Removes deprecated method Piwik\Plugins\API\API::getLastDate()

sgiehl committed:
Removes UserSettingsBC tests

tsteur commented:
We should keep `getListHooksRegistered ` for Piwik 3 👍 

Some tests are failing. Once merged we need to release new versions of the plugins on the Marketplace and require Piwik 3b3

sgiehl committed:
submodule updates

sgiehl commented:
Updated the plugins. Requiring 3b3 in the new plugin versions wasn't required, as the changes should work with all 3.x versions. 
Updated the submodules here. That should fix the tests I guess.

tsteur commented:
True, the new menu methods already existed before so no need to require new version :+1:

sgiehl commented:
Last build, with updated submodules, had "only" failing UI tests. But those are also failing on 3.x-dev branch...

tsteur commented:
Sweet, feel free to merge 👍

mattab commented on plugins/DevicePlugins/DevicePlugins.php:
```diff
@@ -45,11 +44,4 @@ public function addMetricTranslations(&$translations)
-
-    public function renameUserSettingsModuleAndAction(&$module, &$action)
-    {
-        if ($module == 'UserSettings' && $action == 'getPlugin') {
```
@sgiehl why removing this as well, as it would break Embed widgets and is not needed to deprecate IMHO

mattab commented on plugins/UserLanguage/UserLanguage.php:
```diff
@@ -23,7 +23,6 @@ public function registerEvents()
     {
         return array(
             'Live.getAllVisitorDetails'              => 'extendVisitorDetails',
-            'Request.getRenamedModuleAndAction' => 'renameUserSettingsModuleAndAction',
```
idem for all `'Request.getRenamedModuleAndAction'` listeners

tsteur commented on plugins/DevicesDetection/DevicesDetection.php:
```diff
@@ -23,7 +23,6 @@ public function registerEvents()
     {
         return array(
             'Live.getAllVisitorDetails' => 'extendVisitorDetails',
-            'Request.getRenamedModuleAndAction' => 'renameUserSettingsModuleAndAction',
```
We had marked them as to be removed from Piwik 3

mattab replied:
Not sure why it was marked that way, as IMO there is no value to removing these calls. there is no code complexity involved for us, but as a result we would  force people to change some URLs deep into their app or product... which would be time consuming for them

mattab replied:
created https://github.com/piwik/piwik/issues/10831 to follow up
